### DISCUSSION TITLE
can not collect test without console, what should I do?

### DISCUSSION THREAD

**Discussion by echo1nshadow** [pytest-dev/pytest#7900] [Q&A]
I used pyinstaller to package my program, with "-w" option.
program run but can not collect test cases.
If i package my program without "-w" option, it works.
And  i don't want a console , what should i do?

**Comment by The-Compiler** [comment]
What do you mean with "can not collect test cases"? Can you show some code or more details about what you're doing exactly?"

**Comment by nicoddemus** [comment]
The description is lacking, but I _think_ I know what this might be about, because at @ESSS we use pyinstaller to package pytest and run tests from a Windows executable.

When you create an executable without a Console, pytest will break with the default capture method because there's no `fd` console to attach to. Try passing `--capture=sys` to pytest.

(Again this is a shot in the dark, might be about something completely different).

**Reply by echo1nshadow** [reply]
thanks for reply.
I tried passing ``--capture=sys`` to pytest, but it did not work.

And I fixed it by adding code:
``f_handler_out = open("./out.log", 'a')``
`` sys.stdout = f_handler_out``

**Reply by nicoddemus** [reply]
Strange, what you are doing here is basically what `--capture=sys` does. How are you executing pytest inside your executable?
